### PR TITLE
New Timeline Handles + Fixes

### PR DESCRIPTION
**Pull request by shaahmad**
**Live on [Nightingale](https://nightingale-owid.netlify.app/grapher/age-dependency-ratio-old?tab=chart&year=1983&time=1979..2010).** 

Also fixes Timeline grab glitch:
ezgif-1-377a0a7f82bb

**Before:**
image

**After:**
image

### CODE REVIEW

File: charts/HTMLTimeline.tsx [hunk 1]
```diff
@@ -449,18 +451,24 @@ export class Timeline extends React.Component<TimelineProps> {
...
+                        style={{
+                            left: `${startYearProgress * 100 -
+                                this.HANDLE_RADIUS_PERCENTAGE_OF_SLIDER}%`
+                        }}
                     />
                     <div
                         className="interval"
                         style={{
-                            left: `${startYearProgress * 100}%`,
+                            left: `${startYearProgress * 100 - 1.8}%`,
```
Comment (breck7) [inline charts/HTMLTimeline.tsx:462]: nit: what is the 1.8?

File: charts/client/chart.scss [hunk 2]
```diff
@@ -870,22 +870,29 @@ $zindex-IndicatorDropdown: 50;
...
         .handle {
-            height: 1em;
-            border-radius: 0.2em;
-            width: $handle-width;
+            height: $handle-diameter;
+            width: $handle-diameter;
+            border-radius: 100%;
             background: $controls-color;
             position: absolute;
             cursor: col-resize;
```
Comment (breck7) [inline charts/client/chart.scss:883]: Would `cursor: grab` or `cursor: grabbing` be better with the new handles?

Comment (shaahmad) [reply charts/client/chart.scss:884]: Ooh I'm in favor. @vanwoerden veto if needed

File: charts/HTMLTimeline.tsx (hunk 1 above)
Comment (shaahmad) [reply charts/HTMLTimeline.tsx]: threw it out

File: charts/client/chart.scss (hunk 2 above)
Comment (shaahmad) [reply charts/client/chart.scss]: Went ahead and did it for now.

**Comment by shaahmad** [comment]
New updates:
- Handles are 16px, track is 4px, selected interval 6px
- new track color
- odd empty space before end year fix
- mobile years are shortened
- found a way to have both continuous timeline dragging and no glitch when grabbing a handle.

Updating Nightingale now.

File: charts/ChartConfig.tsx [hunk 3]
```diff
@@ -423,9 +423,12 @@ export class ChartConfig {
...
     }
 
-    @computed get formatYearFunction() {
+    @computed get formatYearFunction(): (
+        year: number,
+        options?: { format?: string }
+    ) => string {
         return this.table.hasDayColumn
-            ? (day: number) => formatDay(day)
+            ? (year: number, options?) => formatDay(year, options)
```
Comment (MarcelGerber) [inline charts/ChartConfig.tsx:431]: You can just return the function directly and don't need an additional lambda here.

```suggestion
            ? formatDay
```

File: charts/client/chart.scss [hunk 4]
```diff
@@ -861,32 +862,41 @@ $zindex-IndicatorDropdown: 50;
...
-            height: 1em;
-            border-radius: 0.2em;
-            width: $handle-width;
+            height: $handle-diameter;
+            width: $handle-diameter;
+            border-radius: 100%;
             background: $controls-color;
             position: absolute;
-            cursor: col-resize;
+
```
Comment (MarcelGerber) [inline charts/client/chart.scss:884]: You can just add `margin-left: -$handle-diameter / 2;` here and won't need the whole computing of sizes in JS!

**Review by MarcelGerber — APPROVED** [review]
Other than the few comments this looks good to me.

File: charts/ChartConfig.tsx (hunk 3 above)
Comment (shaahmad) [reply charts/ChartConfig.tsx]: whoa!

File: charts/client/chart.scss (hunk 4 above)
Comment (shaahmad) [reply charts/client/chart.scss:884]: Absolutely wonderful. Removed a lot of ugly code with that.

File: charts/client/chart.scss [hunk 5]
```diff
@@ -861,32 +862,42 @@ $zindex-IndicatorDropdown: 50;
...
-            height: 1em;
-            border-radius: 0.2em;
-            width: $handle-width;
+            height: $handle-diameter;
+            width: $handle-diameter;
+            border-radius: 100%;
             background: $controls-color;
             position: absolute;
+
+            background: #ffffff;
```
Comment (MarcelGerber) [inline charts/client/chart.scss:885]: I'm just now seeing this: There are two background colors defined for the handle. That's one more than we need ;)

**Pull request merged**